tests/data/dataset_readers/wikitables/wikitables_dataset_reader_test.py

```diff
@@ -11,17 +12,244 @@ def test_reader_reads(self):
         dataset = reader.read("tests/fixtures/data/wikitables/sample_data.examples")
         assert len(dataset.instances) == 2
         instance = dataset.instances[0]
-        action_sequence = instance.fields["target_action_sequences"].field_list[0]
-        actions = [l.label for l in action_sequence.field_list]
         question_tokens = ["what", "was", "the", "last", "year", "where", "this", "team", "was",
                            "a", "part", "of", "the", "usl", "a", "-", "league", "?"]
         assert [t.text for t in instance.fields["question"].tokens] == question_tokens
-        assert actions == ['@@START@@', 'd', 'd -> [<d,d>, d]', '<d,d> -> max', 'd -> [<e,d>, e]',
+        entities = instance.fields['table'].knowledge_graph.get_all_entities()
+        assert len(entities) == 47
+        assert 'fb:row.row.year' in entities
+
+        # The content of this will be tested indirectly by checking the actions; we'll just make
+        # sure we get a WikiTablesWorld object in here.
+        assert isinstance(instance.fields['world'].as_tensor({}), WikiTablesWorld)
+
+        actions = [action_field.rule for action_field in instance.fields['actions'].field_list]
+        assert len(actions) == 171
+
+        # This is going to be long, but I think it's worth it, to be sure that all of the actions
+        # we're expecting are present, and there are no extras.
+        assert sorted(actions) == [
+                # Placeholder types
+                "<#1,#1> -> !=",
+                "<#1,#1> -> fb:type.object.type",
+                "<#1,#1> -> var",
+                "<#1,<#1,#1>> -> and",
+                "<#1,<#1,#1>> -> or",
+                "<#1,d> -> count",
+                "<<#1,#2>,<#2,#1>> -> reverse",
+
+                # These complex types are largely just here to build up a few specific functions:
+                # argmin and argmax.
+                "<<d,d>,d> -> [<d,<<d,d>,d>>, d]",
+                "<<d,e>,e> -> [<e,<<d,e>,e>>, e]",
+                "<<d,p>,p> -> [<p,<<d,p>,p>>, p]",
+                "<<d,r>,r> -> [<r,<<d,r>,r>>, r]",
+                "<d,<<d,d>,d>> -> [<d,<d,<<d,d>,d>>>, d]",
+                "<d,<d,<#1,<<d,#1>,#1>>>> -> argmax",
+                "<d,<d,<#1,<<d,#1>,#1>>>> -> argmin",
+                "<d,<d,<<d,d>,d>>> -> [<d,<d,<#1,<<d,#1>,#1>>>>, d]",
+                "<d,<d,d>> -> -",
+                "<d,<e,<<d,e>,e>>> -> [<d,<d,<#1,<<d,#1>,#1>>>>, d]",
+                "<d,<p,<<d,p>,p>>> -> [<d,<d,<#1,<<d,#1>,#1>>>>, d]",
+                "<d,<r,<<d,r>,r>>> -> [<d,<d,<#1,<<d,#1>,#1>>>>, d]",
+
+                # Operations that manipulate numbers.
+                # TODO(mattg): question for Pradeep: why aren't these two-argument functions?
```

Comment:
They take a number or a date (or a set of numbers), and return the set of ALL dates or numbers that are less than, or greater than the given set. I think the executor determines the set of ALL dates or numbers based on the context. For example, `(fb:row.row.start_date (>= (date 2000 -1 -1)))` returns all the rows where the year in `start_date` is later than `2000`.

Reply:
Thanks.